lib/Sema/ConstraintSystem.cpp

```diff
@@ -928,6 +928,82 @@ TypeVariableType *ConstraintSystem::isRepresentativeFor(
   return *member;
 }
 
+static Optional<std::pair<VarDecl *, Type>>
+getPropertyWrappersInformationFromOverload(
+    SelectedOverload resolvedOverload, DeclContext *DC,
+    llvm::function_ref<Optional<std::pair<VarDecl *, Type>>(VarDecl *)>
+        getInformation) {
+  if (resolvedOverload.choice.isDecl()) {
+    if (auto *decl = dyn_cast<VarDecl>(resolvedOverload.choice.getDecl())) {
+      if (auto declInformation = getInformation(decl)) {
+        Type type;
+        VarDecl *memberDecl;
+        std::tie(memberDecl, type) = *declInformation;
+        if (Type baseType = resolvedOverload.choice.getBaseType()) {
+          type = baseType->getTypeOfMember(DC->getParentModule(), memberDecl,
+                                           type);
+        }
+        return std::make_pair(decl, type);
+      }
+    }
+  }
+  return None;
+}
+
+static Optional<std::pair<VarDecl *, Type>>
+getStorageWrapperInformationFromDecl(VarDecl *decl) {
+  if (!decl->hasAttachedPropertyWrapper())
+    return None;
+
+  auto storageWrapper = decl->getPropertyWrapperStorageWrapper();
+  if (!storageWrapper)
+    return None;
+
+  return std::make_pair(storageWrapper, storageWrapper->getInterfaceType());
+}
+
+Optional<std::pair<VarDecl *, Type>>
+ConstraintSystem::getStorageWrapperInformation(
+    SelectedOverload resolvedOverload) {
+  return getPropertyWrappersInformationFromOverload(
+      resolvedOverload, DC,
+      [](VarDecl *decl) { return getStorageWrapperInformationFromDecl(decl); });
+}
+
+static Optional<std::pair<VarDecl *, Type>>
+getPropertyWrapperInformationFromDecl(VarDecl *decl) {
```

Comment:
Let's inline this method into `getPropertyWrapperInformation`?

Reply:
Because of the `None` inline requires an explicit return on the `closure` ... it is ok?

Reply:
That’s totally fine.

Reply:
Right, just fixed :)